diff --git a/sites/all/modules/geofield/geofield.widgets.inc b/sites/all/modules/geofield/geofield.widgets.inc
index e86eaa0..7e28c6a 100644
--- a/sites/all/modules/geofield/geofield.widgets.inc
+++ b/sites/all/modules/geofield/geofield.widgets.inc
@@ -214,6 +214,7 @@ function geofield_field_widget_form(&$form, &$form_state, $field, $instance,
  */
 
 function geofield_widget_openlayers_afterbuild($element, &$form_state) {
+  $settings = $form_state['field'][$element['#field_name']]['und']['instance']['widget']['settings'];
   drupal_add_js(
     array('geofield' => array(
       'data_storage' => (!empty($settings['data_storage'])) ? $settings['data_storage'] : 'collection',
